apps/openmw/mwgui/container.cpp

```diff
@@ -229,11 +229,14 @@ namespace MWGui
                 const ItemStack& item = mModel->getItem(i);
 
                 if (!onTakeItem(item, item.mCount))
-                    break;
-
-                mModel->moveItem(item, item.mCount, playerModel);
+                {
+                    MWBase::Environment::get().getWindowManager()->removeGuiMode(GM_Container);
+                    return;
+                }
```

Comment:
Looks like there is a change in behavior here. If you 'take all' with the pickpocket window, successfully roll the first few items but fail on a later one, no items are added to your inventory. Is this correct?

Reply:
Sorry, my fault. How can I workaround this bug?

Reply:
As a simple solution, I can just add equipReplacement flag to ItemModel::moveItem() and ItemModel::removeItem().
Then I will rollback changes in container.cpp.

Reply:
I'd prefer not complicating the item model API for something that doesn't apply to all item models.

How about this?
<pre>
if container is NPC:
   for every equipped item in model:
      unequip item
   update model
take all items
</pre>

It's still hack-ish, but at least the impact is limited to one file.

Reply:
Hmm... How is this code supposed to help us? And I did not get where we should place it.
We have no problems with **unequipping**, we have a problem with **autoequipping**.
When we use for() loop to remove all items one by one, we should not touch items stacks (as autoEquip does).
As a simpliest, but a dumb solution, we can use a while() loop instead of for(), and update model after every removed stack.
But in this case we can get an endless loop, if something will go wrong, so we will have to add some failsafe break.

Reply:
> How is this code supposed to help us? 

By un-equipping everything first, the problematic code path in InventoryStore::remove() that re-equips an alternative item when an equipped item is removed should no longer trigger (because the removed item wasn't equipped). Makes sense?

> while() loop instead of for(), and update model after every removed stack.

I've thought about this. The quadratic complexity could be a problem, though (i.e. for every item, we need to update the model, looking at every item again). How fast is this for a container with say, 1000 items? Good point with the failsafe, too.

Reply:
> Makes sense?

Yes. It is implemented now.